great_tables/_formats.py

```diff
@@ -2716,22 +2707,28 @@ def _normalize_locale(locale: Union[str, None] = None) -> Union[str, None]:
     supplied_locale = _str_replace(locale, "_", "-")
 
     # Resolve any default locales into their base names (e.g., 'en-US' -> 'en')
-    if supplied_locale in _get_default_locales_list():
-        default_locales = _get_default_locales_data()
-        resolved_locale = default_locales[
-            default_locales["default_locale"] == supplied_locale
-        ].iloc[0]["base_locale"]
-
-        # Ensure that `resolved_locale` is of the type 'str'
-        resolved_locale: Any
-        if not isinstance(resolved_locale, str):
-            raise TypeError(
-                "Variable type mismatch. Expected str, got something entirely different."
-            )
-    else:
-        resolved_locale = supplied_locale
+    # TODO: remove pandas
+    default_locales = _get_default_locales_data()
 
-    return resolved_locale
+    matches = [
+        entry["base_locale"]
+        for entry in default_locales
+        if entry["default_locale"] == supplied_locale
+    ]
+
+    if matches:
+        return matches[0]
+
+    try:
```

Comment:
This is a new behavior. Before it allowed unrecognized to pass through. I added some babel validation, to try and only let the subset of things through that:

* aren't in our locale data
* will end up working with babel

(but could have gotten it wrong!)